docs/glossary.md

```diff
@@ -0,0 +1,181 @@
+"named branches" to distinguish them from
+[anonymous branches](#anonymous-branch), but note that they are more similar
+to Git's branches than to
+[Mercurial's named branches](https://www.mercurial-scm.org/wiki/Branch#Named_branches).
+
+See [here](branches.md) for details.
+
+## Change
+
+A change is a commit as it [evolves over time](#rewrite).
+
+## Change ID
+
+A change ID is a unique identifier for a [change](#change). They are typically
+16 bytes long and are often randomly generated. They are presented as
+hexadecimal numbers at the start of the line in `jj log`, using 12 hexadecimal
+digits by default. However, the hexadecimal digits are taken from the end of the
+alphabet (z-k) instead of the usual 0-9 and a-f.
+
+## Commit
+
+A snapshot of the files in the repository at a given point in time (technically
+a [tree object](#tree)), together with some metadata. The metadata includes the
+author, the date, and pointers to the commit's parents. Through the pointers to
+the parents, the commits form a
+[Directed Acyclic Graph (DAG)](https://en.wikipedia.org/wiki/Directed_acyclic_graph)
+.
+
+Note that even though commits are stored as snapshots, they are often treated
+as differences between snapshots, namely compared to their parent's snapshot. If
+they have more than one parent, then the difference is computed against the
+result of merging the parents. For example, `jj diff` will show the differences
+introduced by a commit compared to its parent(s), and `jj rebase` will apply
+those changes onto another base commit.
+
+The word "revision" is used as a synonym for "commit".
+
+## Commit ID
+
+A commit ID is a unique identifier for a [commit](#commit). They are 20 bytes
+long when using the Git backend. They are presented in regular hexadecimal
+format at the end of the line in `jj log`, using 12 hexadecimal digits by
+default.
+
+## Co-located repos
+
+When using the Git [backend](#backend) and the backing Git repository's `.git/`
+directory is a sibling of `.jj/`, the we call the repository "co-located".
+
+See [here](git-compatibility.md#co-located-jujutsugit-repos) for details.
+
+## Conflict
+
+TODO
+
+Conflicts can occur in files, but they can also occur in branches, for example.
+
+See [here](conflicts.md) for details.
+
+## Divergent change
+
+A divergent change is a [change](#change) that has more than one visible commit.
+
+## Operation
+
+A snapshot of the [visible commits](#visible-commits) and [branches](#branches)
+at a given point in time (technically a [view object](#view)), together with
+some metadata. The metadata includes the username, hostname, timestamps, and
+pointers to the operation's parents.
+
+## Operation log
+
+The [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) formed by
+[operation](#operation) objects, much in the same way that commits form a DAG,
+which is sometimes called the "commit history".
+
+## Repository
+
+Basically everything reachable from the [operation log](#operation-log), i.e.
```

Comment:
Hm, does this match how most people use the word? I think people typically refer to everything on disk (e.g. the storage backend, working copy) as the "repository".

e.g. "My repository is corrupted, better delete it".

Reply:
I don't think of the working copy as part of the repository, but you may be right that most people do. I've rephrased it as "everything under `.jj/`", so it's at least not the same as the definition of "workspace".

Reply:
Sounds good. Have you pushed that change?

Reply:
Now I have :)